resolver-dns/src/test/java/io/netty/resolver/dns/SearchDomainTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import org.junit.After;
+import org.junit.Test;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+public class SearchDomainTest {
+
+  private DnsNameResolverBuilder newResolver() {
+    return new DnsNameResolverBuilder(group.next())
+        .channelType(NioDatagramChannel.class)
+        .nameServerAddresses(DnsServerAddresses.singleton(dnsServer.localAddress()))
+        .maxQueriesPerResolve(1)
+        .optResourceEnabled(false);
+  }
+
+  private TestDnsServer dnsServer;
+  private final EventLoopGroup group = new NioEventLoopGroup(1);
```

Comment:
newResolver() is called with different search domains configurations in various tests.